crates/ra_assists/src/handlers/merge_imports.rs

```diff
@@ -72,9 +72,24 @@ fn try_merge_trees(old: &ast::UseTree, new: &ast::UseTree) -> Option<ast::UseTre
     let lhs = old.split_prefix(&lhs_prefix);
     let rhs = new.split_prefix(&rhs_prefix);
 
+    let mut should_insert_comma = true;
+
+    lhs.syntax()
+        .last_child()
+        .and_then(|child| child.children().last())
+        .and_then(|last| last.next_sibling_or_token())
+        .map(|next_sibling| {
+            // FIXME: there's probably a better way to check if it's a COMMA
+            if let NodeOrToken::Token(maybe_comma) = next_sibling {
+                should_insert_comma = maybe_comma.to_string() != ",";
+            }
+        });
+
```

Comment:
```

    let should_insert_comma = lhs
        .syntax()
        .last_child()
        .and_then(|child| child.children().last())
        .and_then(|last| last.next_sibling_or_token())
        .map(|it| it.kind() != T![,])
        .unwrap_or(true);
```

Reply:
And here is a more correct one which skip white spaces properly :
```rust
let should_insert_comma = lhs
    .use_tree_list()?
    .r_curly_token()
    .and_then(|it| skip_trivia_token(it.prev_token()?, Direction::Prev))
    .map(|it| it.kind() != T![,])
    .unwrap_or(true);
```

Reply:
Thanks, this is much better!